package.json

```diff
@@ -36,21 +37,53 @@
       ],
       "configuration": "./language-configuration.json"
     }],
-    "configurationDefaults": {
-      "[lc2k]": {
-        "editor.insertSpaces": false,
-        "editor.tabSize": 8,
-        "editor.autoIndent": false,
-        "editor.formatOnSave": false
-      }
-    },
+
     "grammars": [{
       "language": "lc2k",
       "scopeName": "source.lc2k",
       "path": "./syntaxes/lc2k.tmLanguage.json"
     }],
     "capabilities": {
       "definitionProvider": "true"
+    },
+    "configuration": {
+      "type": "object",
+      "title": "LC2K configuration",
+      "properties": {
+        "lc2k.format.enable": {
+          "type": "boolean",
+          "default": true,
+          "description": "Enables this extention to format .as files."
+        },
+        "lc2k.format.addNewlineAtEOF": {
+          "type": "boolean",
+          "default": true,
+          "description": "Add a newline at the end of a file if there isn't one."
+        },
+        "lc2k.format.filloffset": {
```

Comment:
Should these be styled like `fillOffset` instead?

Reply:
Mention the `-1` to disable that's described above?

Reply:
Maybe use `null` instead of `-1`, since lots of other config things seem to use `null` for that purpose.